src/Hosting/Hosting/src/WebHostBuilderExtensions.cs

```diff
@@ -61,6 +61,33 @@ private static IWebHostBuilder Configure(this IWebHostBuilder hostBuilder, Actio
             });
         }
 
+        /// <summary>
+        /// Specify a factory that creates the startup instance to be used by the web host.
+        /// </summary>
+        /// <param name="hostBuilder">The <see cref="IWebHostBuilder"/> to configure.</param>
+        /// <param name="startupFactory">A delegate that specifies a factory for the startup class.</param>
+        /// <returns>The <see cref="IWebHostBuilder"/>.</returns>
+        public static IWebHostBuilder UseStartup(this IWebHostBuilder hostBuilder, Func<WebHostBuilderContext, object> startupFactory)
+        {
```

Comment:
What about the app name?
https://github.com/dotnet/aspnetcore/blob/088380b45cad29f48635867526b954d8c86b84f7/src/Hosting/Hosting/src/WebHostBuilderExtensions.cs#L73-L75

Reply:
Good catch.